group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -727,20 +728,42 @@ private void maybeUpdateSubscribedTopicNames(
      * @param newMember             The new member.
      */
     private static void maybeUpdateSubscribedTopicNames(
+        TopicsImage topics,
         Map<String, Integer> subscribedTopicCount,
         ConsumerGroupMember oldMember,
         ConsumerGroupMember newMember
     ) {
         if (oldMember != null) {
-            oldMember.subscribedTopicNames().forEach(topicName ->
-                subscribedTopicCount.compute(topicName, ConsumerGroup::decValue)
-            );
+            String oldMemberRegex = oldMember.subscribedTopicRegex();
+            if (oldMemberRegex == null || oldMemberRegex.isEmpty()) {
+                oldMember.subscribedTopicNames().forEach(topicName ->
+                        subscribedTopicCount.compute(topicName, ConsumerGroup::decValue)
+                );
+            } else {
+                Pattern pattern = Pattern.compile(oldMemberRegex);
+                oldMember.subscribedTopicNames()
```

Comment:
Do we need this condition? Eventually for the old member's subscribed topic we need to decrement the counts. Assuming that any topic subscribed by virtue of regex based subscriptions as well will increment the count, we don't need to run pattern matching like this I feel.